server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Message.java

```diff
@@ -244,11 +244,15 @@ protected static boolean areAttachedMessagesKeysInAttachments(ImmutableList<Atta
 
     private static boolean hasAttachment(List<Attachment> attachments) {
         return attachments.stream()
-                .filter(attachment -> !attachment.isIsInline())
+                .filter(attachment -> isNotRealInline(attachment))
                 .findAny()
                 .isPresent();
     }
 
+    private static boolean isNotRealInline(Attachment attachment) {
+        return !attachment.isIsInline() || (attachment.isIsInline() && !attachment.getCid().isPresent());
```

Comment:
Hmmmmm

Still not readable yet.

I would propose : 

```
return !attachment.isIsInline() || isInlinedWithoutCid(attachment);
```

With : 

```
private static boolean isInlinedWithoutCid(Attachment attachment) {
    return attachment.isIsInline() && !attachment.getCid().isPresent();
}
```

Moreover **isNotRealInline** is a bad method name. And this is code duplication. Detecting if this is a "real attachment" should be a concern of Attachment class. Please move logic for this in a method called **isInlinedWithCid**, and write test for it.